### PR TITLE
Khwe 7 my songs view

### PR DESCRIPTION
**Pull request by kozak-iz-kh**
screenshot_16

### CODE REVIEW

**Comment by ihor4568** [comment]
this table does not look like the one here https://getmdl.io/components/index.html#tables-section

please COPY STYLES, just make them in BEM-manner

**Comment by ihor4568** [comment]
please remove title from 3 dots column and add 3 dots button there (just button with 3 dots icon)

File: src/components/MySongsView/MySongsViewTable/MySongsViewTable.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<table class="mySongsTable">
```
Comment (ihor4568) [inline src/components/MySongsView/MySongsViewTable/MySongsViewTable.html]: please use kebab case

File: src/components/MySongsView/MySongsViweRow/MySongsViewRow.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<td class="mySongsTable__td mySongsTable__td_btnPlace">
```
Comment (ihor4568) [inline src/components/MySongsView/MySongsViweRow/MySongsViewRow.html]: not sure that this is worth extracting as separate component

File: src/components/MySongsView/MySongsViweRow/MySongsViewRow.scss [hunk 3]
```diff
@@ -0,0 +1,70 @@
+.mySongsTable__td {
```
Comment (ihor4568) [inline src/components/MySongsView/MySongsViweRow/MySongsViewRow.scss]: please use kebab case

File: src/components/MySongsView/MySongsViweRow/MySongsViewRow.scss [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  height: 3rem;
```
Comment (ihor4568) [inline src/components/MySongsView/MySongsViweRow/MySongsViewRow.scss]: I do not see proper usage of BEM methodology

File: src/index.scss [hunk 5]
```diff
@@ -9,3 +9,6 @@
 @import "./components/Main/Main";
 @import "./components/Player/Player";
 @import "./components/Page/Page";
+
+@import "./components/MySongsView/MySongsViweRow/MySongsViewRow";
+@import "./components/MySongsView/MySongsViewTable/MySongsViewTable.scss";
```
Comment (ihor4568) [inline src/index.scss]: remove extension here

File: src/components/MySongsView/MySongsViewTable/MySongsViewTable.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      <th class="mySongsTable__th mySongsTable__th_duration">
+        <i class="material-icons">watch_later</i>
+      </th>
+      <th class="mySongsTable__th mySongsTable__th_artist">Artist</th>
+      <th class="mySongsTable__th mySongsTable__th_album">Album</th>
+      <th class="mySongsTable__th mySongsTable__th_3dots">3dot</th>
+    </tr>
+  </thead>
+  <tbody class="mySongsTable__body">
+    <tr class="mySongsTable__row mySongsTable__row1"></tr>
```
Comment (ihor4568) [inline src/components/MySongsView/MySongsViewTable/MySongsViewTable.html]: how are you supposed to display several rows? please use loops or foreach and display data from array

File: src/components/Main/Main.html [hunk 7]
```diff
@@ -2,7 +2,7 @@
   <aside class="main__sidebar mdc-drawer mdc-drawer--dismissible">
     <div class="mdc-drawer__content">
       <div class="mdc-list main__list">
-        <a class="mdc-list-item mdc-list-item--activated" href="#" aria-selected="true">
+        <a class="mdc-list-item mdc-list-item--activated main__mySongs-link" href="#" aria-selected="true">
```
Comment (ihor4568) [inline src/components/Main/Main.html]: please use kebab-case for class names

File: src/components/MySongsTable/MySongsTable.html [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        <i class="material-icons my-songs-table__th_duration-icon">watch_later</i>
+      </th>
+      <th class="my-songs-table__th my-songs-table__th_artist">Artist</th>
+      <th class="my-songs-table__th my-songs-table__th_album">Album</th>
+      <th class="my-songs-table__th my-songs-table__th_3dots"></th>
+    </tr>
+  </thead>
+  <tbody class="my-songs-table__body">
+    <!-- one hardcoded row -->
+    <tr class="my-songs-table__row">
```
Comment (ihor4568) [inline src/components/MySongsTable/MySongsTable.html]: why do you need one hardcoded row?

File: src/components/MySongsTable/MySongsTable.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  mount() {
+    this.mountPoint.innerHTML = this.render();
+  }
+
+  render() {
+    return mySongsTableTemplate(this.props);
```
Comment (ihor4568) [inline src/components/MySongsTable/MySongsTable.js]: please specify what exactly from props you are passing to the template

File: src/components/MySongsTable/MySongsTable.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import mySongsTableTemplate from "./MySongsTable.html";
+export class MySongsTableComponent {
```
Comment (ihor4568) [inline src/components/MySongsTable/MySongsTable.js:31]: please add space between import declaration and class decl

File: src/components/MySongsTable/MySongsTable.scss [hunk 11]
```diff
@@ -0,0 +1,126 @@
+.my-songs-table {
+  width: calc(100% - 5rem);
```
Comment (ihor4568) [inline src/components/MySongsTable/MySongsTable.scss]: why width is not 100%?

File: src/components/MySongsTable/MySongsTable.scss [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+  white-space: nowrap;
+  font-size: 0.8rem;
+  background-color: #fff;
+  color: #757575;
+  display: table;
+
+  box-shadow: 0 0.1rem 0.1rem 0 rgba(0, 0, 0, 0.14),
+    0 0.2rem 0.1rem -0.15rem rgba(0, 0, 0, 0.2),
+    0 0.1rem 0.3rem 0 rgba(0, 0, 0, 0.12);
+  font-family: "Roboto", "Helvetica", "Arial", sans-serif !important;
```
Comment (ihor4568) [inline src/components/MySongsTable/MySongsTable.scss]: please use default font family

File: src/components/MySongsTable/MySongsTable.html [hunk 13]
```diff
@@ -0,0 +1,69 @@
+<table class="my-songs-table">
+  <caption class="mdc-typography--headline4 my-songs-table__caption">My Songs</caption>
+  <thead>
+    <tr>
+      <th class="my-songs-table__th"></th>
+      <th class="my-songs-table__th"></th>
+      <th class="my-songs-table__th my-songs-table__th_name ">Name</th>
```
Comment (ihor4568) [inline src/components/MySongsTable/MySongsTable.html]: please remove extra space at the end of `class`

File: src/components/Main/Main.html (hunk 7 above)
Comment (kozak-iz-kh) [reply src/components/Main/Main.html]: resolved

File: src/components/MySongsTable/MySongsTable.html (hunk 8 above)
Comment (kozak-iz-kh) [reply src/components/MySongsTable/MySongsTable.html]: resolved. deleted hard coded row

File: src/components/MySongsTable/MySongsTable.js (hunk 9 above)
Comment (kozak-iz-kh) [reply src/components/MySongsTable/MySongsTable.js]: resolved. specified props

File: src/components/MySongsTable/MySongsTable.js (hunk 10 above)
Comment (kozak-iz-kh) [reply src/components/MySongsTable/MySongsTable.js:31]: resolved

File: src/components/Main/Main.js [hunk 14]
```diff
@@ -3,14 +3,42 @@ import { MDCDrawer } from "@material/drawer";
...
+    artist: "Folk",
+    album: "Super"
+  }
+];
+
 export class MainComponent {
   constructor(mountPoint, props = {}) {
     this.mountPoint = mountPoint;
     this.props = props;
+    this.tableData = TABLE_DATA;
```
Comment (ghost) [inline src/components/Main/Main.js]: pass `TABLE_DATA` into the props or simply access it as `TABLE_DATA.`

File: src/components/MySongsTable/MySongsTable.scss [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    &_album {
+      text-align: left;
+    }
+  }
+
+  &__td {
+    position: relative;
+    // height: 3rem;
```
Comment (ghost) [inline src/components/MySongsTable/MySongsTable.scss]: don't forget to remove it pls

**Review by ihor4568 — APPROVED** [review]

**Pull request merged**